MekHQ/src/mekhq/gui/dialog/ImageChoiceDialog.java

```diff
@@ -478,99 +382,39 @@ public String getFileName() {
-                iconMap.put(IconPackage.FORCE_TYPE, tmp);
-            }
-        }
-        if (tableFormations.getSelectedRow() == -1) {
-            iconMap.remove(IconPackage.FORCE_FORMATIONS, iconMap.get(IconPackage.FORCE_FORMATIONS));
-        } else {
-            tmp = new Vector<>();
-            for (int index : tableFormations.getSelectedRows()) {
-                tmp.add((String) tableFormations.getValueAt(index, 0));
-                iconMap.put(IconPackage.FORCE_FORMATIONS, tmp);
-            }
-        }
-        if (tableAdjustments.getSelectedRow() == -1) {
-            iconMap.remove(IconPackage.FORCE_ADJUSTMENTS, iconMap.get(IconPackage.FORCE_ADJUSTMENTS));
-        } else {
-            tmp = new Vector<>();
-            for (int index : tableAdjustments.getSelectedRows()) {
-                tmp.add((String) tableAdjustments.getValueAt(index, 0));
-                iconMap.put(IconPackage.FORCE_ADJUSTMENTS, tmp);
-            }
-        }
-        if (tableAlphanumerics.getSelectedRow() == -1) {
-            iconMap.remove(IconPackage.FORCE_ALPHANUMERICS, iconMap.get(IconPackage.FORCE_ALPHANUMERICS));
-        } else {
-            tmp = new Vector<>();
-            for (int index : tableAlphanumerics.getSelectedRows()) {
-                tmp.add((String) tableAlphanumerics.getValueAt(index, 0));
-                iconMap.put(IconPackage.FORCE_ALPHANUMERICS, tmp);
-            }
-        }
-        if (tableSpecialModifiers.getSelectedRow() == -1) {
-            iconMap.remove(IconPackage.FORCE_SPECIAL_MODIFIERS, iconMap.get(IconPackage.FORCE_SPECIAL_MODIFIERS));
-        } else {
-            tmp = new Vector<>();
-            for (int index : tableSpecialModifiers.getSelectedRows()) {
-                tmp.add((String) tableSpecialModifiers.getValueAt(index, 0));
-                iconMap.put(IconPackage.FORCE_SPECIAL_MODIFIERS, tmp);
-            }
-        }
-        if (tableBackgrounds.getSelectedRow() == -1) {
-            iconMap.remove(IconPackage.FORCE_BACKGROUNDS, iconMap.get(IconPackage.FORCE_BACKGROUNDS));
-        } else {
-            tmp = new Vector<>();
-            for (int index : tableBackgrounds.getSelectedRows()) {
-                tmp.add((String) tableBackgrounds.getValueAt(index, 0));
-                iconMap.put(IconPackage.FORCE_BACKGROUNDS, tmp);
-            }
-        }
-        if (tableLogos.getSelectedRow() == -1) {
-            iconMap.remove(IconPackage.FORCE_LOGOS, iconMap.get(IconPackage.FORCE_LOGOS));
-        } else {
-            tmp = new Vector<>();
-            for (int index : tableLogos.getSelectedRows()) {
-                tmp.add((String) tableLogos.getValueAt(index, 0));
-                iconMap.put(IconPackage.FORCE_LOGOS, tmp);
+        Vector<String> temp;
+        LayeredForceIcon[] layeredForceIcons = LayeredForceIcon.values();
+        for (int i = 0; i < layeredTables.length; i++) {
+            if (layeredTables[i].getSelectedRow() <= 0) {
```

Comment:
A comment as to why this is `<=` (no selection or the first) would be good.

Reply:
Added.